applications/app/controllers/SignupPageController.scala

```diff
@@ -15,20 +16,31 @@ class SignupPageController(
     wsClient: WSClient,
     val controllerComponents: ControllerComponents,
     csrfAddToken: CSRFAddToken,
+    newsletterSignupAgent: NewsletterSignupAgent,
 )(implicit context: ApplicationContext)
     extends BaseController
-    with ImplicitControllerExecutionContext {
+    with ImplicitControllerExecutionContext
+    with GuLogging {
 
   val defaultCacheDuration: Duration = 15.minutes
 
   def renderNewslettersPage(): Action[AnyContent] =
     csrfAddToken {
       Action { implicit request =>
-        Cached(defaultCacheDuration)(
-          RevalidatableResult.Ok(
-            NewsletterHtmlPage.html(StaticPages.simpleNewslettersPage(request.path)),
-          ),
-        )
+        val groupedNewsletters: Either[String, GroupedNewslettersResponse] =
+          newsletterSignupAgent.getGroupedNewsletters()
+
+        groupedNewsletters match {
+          case Right(groupedNewsletters) =>
+            Cached(defaultCacheDuration)(
+              RevalidatableResult.Ok(
+                NewsletterHtmlPage.html(StaticPages.simpleNewslettersPage(request.path, groupedNewsletters.toList())),
+              ),
+            )
+          case Left(e) =>
+            log.error(s"API call to get newsletters failed: $e")
+            Cached(15.minute)(WithoutRevalidationResult(InternalServerError))
```

Comment:
I don't think we would generally cache the internal server error, so that we can recover quickly once the underlying issue is resolved, and to serve the error to as few users as possible.

Reply:
So I based it off this:
https://github.com/guardian/frontend/blob/d4c89252effac85868910165c92354c3d7905bed/common/app/controllers/EmailSignupController.scala#L114

What would the return be for this otherwise? I think I'm struggling to fully get my head around it because the definition of `WithoutRevalidationResult` doesn't explain what it does exactly and I'd assumed that it was a way to handle the error such that it served the old content (based mainly off what I thought I'd seen happen on the Email Embeds)

Reply:
Otherwise is this an acceptable approach to avoid caching
```scala
case Left(e) =>
        log.error(s"API call to get newsletters failed: $e")
        NoCache(InternalServerError)
```